lib/navigation/service/app_router_delegates.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:acs_upb_mobile/authentication/service/auth_provider.dart';
+import 'package:acs_upb_mobile/authentication/view/login_view.dart';
+import 'package:acs_upb_mobile/authentication/view/sign_up_view.dart';
+import 'package:acs_upb_mobile/main.dart';
+import 'package:acs_upb_mobile/navigation/model/navigation_state.dart';
+import 'package:acs_upb_mobile/navigation/model/route_paths.dart';
+import 'package:acs_upb_mobile/navigation/view/web_shell.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+abstract class AppRouterDelegate extends RouterDelegate<RoutePath>
+    with ChangeNotifier, PopNavigatorRouterDelegateMixin<RoutePath> {
+  T _initializeProvider<T extends Listenable>(BuildContext context) {
+    final T provider = Provider.of<T>(context)..addListener(notifyListeners);
+    return provider;
+  }
+}
+
+class MainRouterDelegate extends AppRouterDelegate {
+  MainRouterDelegate({@required NavigationStateProvider navigationState})
+      : _navigatorKey = GlobalKey<NavigatorState>(),
+        _navigationState = navigationState {
+    _navigationState.addListener(notifyListeners);
+  }
+
+  final GlobalKey<NavigatorState> _navigatorKey;
+  final NavigationStateProvider _navigationState;
+
+  // Providers
```

Comment:
Unnecessary comment? Plus it's just one provider

Reply:
there actually are two (the first is at line 28) but it is indeed unnecessary